tests/transforms/test_decompositions.py

```diff
@@ -551,3 +534,24 @@ def test_two_qubit_decomposition_jax(self, U, wires):
         )
 
         assert check_matrix_equivalence(U, obtained_matrix, atol=1e-7)
+
+    @pytest.mark.parametrize("wires", [[0, 1], ["a", "b"], [3, 2], ["c", 0]])
+    @pytest.mark.parametrize("U_pair", samples_su2_su2)
+    def test_two_qubit_decomposition_tensor_products_jax(self, U_pair, wires):
+        """Test that a one-qubit matrix in isolation is correctly decomposed."""
+        jax = pytest.importorskip("jax")
+
+        from jax.config import config
+
+        remember = config.read("jax_enable_x64")
+        config.update("jax_enable_x64", True)
+
+        U1 = jax.numpy.array(U_pair[0], dtype=jax.numpy.complex128)
+        U2 = jax.numpy.array(U_pair[1], dtype=jax.numpy.complex128)
+        U = qml.math.kron(U1, U2)
+
+        obtained_decomposition = two_qubit_decomposition(U, wires=wires)
+
+        obtained_matrix = compute_matrix_from_ops_two_qubit(obtained_decomposition, wires)
+
+        assert check_matrix_equivalence(U, obtained_matrix, atol=1e-7)
```

Comment:
Nice! so decomposition works in all autodiff libraries, even if the gradient doesn't?

Reply:
Yup! :tada: